### PR TITLE
Allow for swift's init customization

### PR DESCRIPTION
**Pull request by Ecarrion**
## Summary

https://github.com/quicktype/quicktype/pull/1218 added the ability to add default values to swift code by following the strategy proposed in [this tutorial](https://blog.quicktype.io/customizing-quicktype/).  With that came the need to modify how `quicktype` renders the `init` method of a type because a `let`(inmutable value) can't be initialized twice or the compiler will throw an error.

compiler-error

This PR aims to allow `quicktype-core` users to customize a type's `init` by defining the set of properties that has to be rendered. (Instead of all of them).

## Implementation Details

I've added a `initializableProperties` method which is basically a wrapper  for `forEachClassProperty` method to be used inside the `renderClassDefinition` method when rendering a type's `init`

Additionally I've defined a new ~class~ interface called `SwiftProperty` which is the return type of the newly `initializableProperties` method so consumers can have a concrete type to deal with when overriding it.

## Usage

```typescript
class CustomRenderer extends SwiftRenderer {
    protected initializableProperties(c: ClassType): SwiftProperty[] {
        const properties = super.initializableProperties(c)
        const filteredProperties: SwiftProperty[] = []

        // Filter properties that have a default value
        for (let property of properties) {
            if (this.defaultValueFor(property) === undefined) {
                 filteredProperties.push(property)
            }
        }
        return filteredProperties
    }
}
```

### CODE REVIEW

File: src/quicktype-core/language/Swift.ts [hunk 1]
```diff
@@ -98,6 +98,19 @@ class SwiftDateTimeRecognizer extends DefaultDateTimeRecognizer {
     }
 }
 
+export class SwiftProperty {
```
Comment (schani) [inline src/quicktype-core/language/Swift.ts]: This should just be an `interface`, not a `class`.  It just holds data.

**Comment by schani** [comment]
This doesn't compile:

```
Step 53/54 : RUN npm install --unsafe-perm
--
  | ---> Running in 62ce3426bf72
  |  
  | > elm@0.18.0-exp5 install /app/node_modules/elm
  | > npx binwrap-install
  |  
  |  
  | > quicktype@15.0.0 prepare /app
  | > npm run build
  |  
  |  
  | > quicktype@15.0.0 build /app
  | > script/build.ts
  |  
  | src/quicktype-core/language/Swift.ts(105,15): error TS2304: Cannot find name 'ForEachPosition'.
  | src/quicktype-core/language/Swift.ts(105,15): error TS4031: Public property 'position' of exported class has or is using private name 'ForEachPosition'.
  | src/quicktype-core/language/Swift.ts(106,83): error TS2304: Cannot find name 'ForEachPosition'.
  | src/quicktype-core/language/Swift.ts(106,83): error TS4063: Parameter 'position' of constructor from exported class has or is using private name 'ForEachPosition'.
  | npm ERR! code ELIFECYCLE
  | npm ERR! errno 1
  | npm ERR! quicktype@15.0.0 build: `script/build.ts`
  | npm ERR! Exit status 1
  | npm ERR!
  | npm ERR! Failed at the quicktype@15.0.0 build script.
  | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
  |  
  | npm ERR! A complete log of this run can be found in:
  | npm ERR!     /root/.npm/_logs/2019-05-09T02_59_40_618Z-debug.log
  | npm ERR! code ELIFECYCLE
  | npm ERR! errno 1
  | npm ERR! quicktype@15.0.0 prepare: `npm run build`
  | npm ERR! Exit status 1
  | npm ERR!
  | npm ERR! Failed at the quicktype@15.0.0 prepare script.
  | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
  |  
  | npm ERR! A complete log of this run can be found in:
  | npm ERR!     /root/.npm/_logs/2019-05-09T02_59_40_663Z-debug.log
  | The command '/bin/sh -c npm install --unsafe-perm' returned a non-zero code: 1
```

File: src/quicktype-core/language/Swift.ts (hunk 1 above)
Comment (Ecarrion) [reply src/quicktype-core/language/Swift.ts]: TIL: Interfaces in typescript

**Comment by Ecarrion** [comment]
@schani Please review again 🙏

**Comment by schani** [comment]
Great work!  Thank you so much!

**Pull request merged**